### ISSUE TITLE
reserved identifier violation

### ISSUE THREAD

**Issue by elfring** [lip6/libDDD#38]
I would like to point out that identifiers like “[`_GDDD`](https://github.com/lip6/libDDD/blob/865b64dcc26bc6ec1dd1ea109b7b94384224d558/ddd/DDD.cpp#L53 "Update candidate")” and “[`_GHom`](https://github.com/lip6/libDDD/blob/9531a4aa9557993448f2b5d4f167f4442fb0f1b6/ddd/Hom.cpp#L50 "Another update candidate")” [do eventually not fit](https://wiki.sei.cmu.edu/confluence/display/cplusplus/DCL51-CPP.+Do+not+declare+or+define+a+reserved+identifier "Do not declare an identifier which is reserved for the compiler implementation.") to the expected naming convention of the C++ language standard.
Would you like to adjust your selection for unique names?

**Comment by elfring** [comment]
>I have my own code quality analysis tools, thank you.

:thinking: It seems that we stumble on a temporary disagreement about implementation details.

:thought_balloon: I suggest to avoid that software components [depend on undefined behaviour](https://www.securecoding.cert.org/confluence/display/c/CC.+Undefined+Behavior#CC.UndefinedBehavior-ub_106 "Avoiding undefined behaviour?").

**Comment by yanntm** [comment]
Right, if my *formal verification tool* exhibits UB or some other non deterministic behavior, please do report that, as a user. 

But this code base is stable, and rigorously tested at the functional level (e.g. winning formal verification competitions), so it seems extremely doubtful that any such behavior exists.

+ it does not even target C++ standard since we use gcc specific features in it anyway. 
So if the symbol does not clash with anything in gcc, I'm happy. 

If it did, I *would* already know about it, it would most probably create probably severe compilation errors, the _prefix naming of storage classes w.r.t. to containers is used through the code base. If UB existed w/o compilation errors the functional tests would exhibit it.

So we are back to simply an issue submitted by an automated code quality tool (basically a bot), not a user reporting an actual issue with the software.

A repo with a lot of pending issues just looks bad, so I certainly do not want bots (other than my own) creating issues or PR, esp. if they are in fact irrelevant like this one. Just answering this pointless issue that you don't even care about is taking my time.

So the point remains, issues are not for interacting with automated code quality analysis tools, which are only indicative anyway in the best case. 

So, please refrain from (mis)using this communication vector to pander whatever code quality analysis improvement you're trying to sell, it's bad etiquette.

**Comment by yanntm** [comment]
I'll link a few of my colleagues here so I can share my thoughts with them as the numerous issues submitted all concern MCC related tools. so these are all people I know.


https://github.com/cesaro/cunf/issues/1

https://github.com/TAPAAL/verifytapn/issues/23

https://github.com/greatspn/SOURCES/issues/41

https://github.com/Tj-Cong/EnPAC_2021/issues/1

https://github.com/asminer/smart/issues/1

**Comment by elfring** [comment]
>…, so it seems extremely doubtful that any such behavior exists.

:thought_balloon: Undefined behaviour exists as known rules from the standard specification of the programming language “C++” get violated.


>it does not even target C++ standard since we use gcc specific features in it anyway.

:thought_balloon: I recommend to reconsider also this view.


>So if the symbol does not clash with anything in gcc, I'm happy.

:thinking: I find such a software development approach questionable.


>…, not a user reporting an actual issue with the software.

I am a special user who dared to present another bug report.

**Comment by elfring** [comment]
>so these are all people I know.

:thought_balloon: Several developers and code reviewers are aware that some programmers repeat known weaknesses and mistakes (in various software components).
